include/swift/Basic/SourceManager.h

```diff
@@ -92,6 +92,37 @@ class GeneratedSourceInfo {
   StringRef onDiskBufferCopyFileName = StringRef();
 };
 
+/// Stores range information for a \c #if block in a source buffer.
+class IfConfigRangeInfo final {
+  /// The range of the entire \c #if block, including \c #else and \c #endif.
+  CharSourceRange WholeRange;
+
+  /// The range of the active selected body, if there is one.
```

Comment:
Does the range include `#if` (or equivalent as well) or just the actual statements/expressions/declarations? Also, I assume that an invalid active range represents that there is no active branch, right? Might be worth pointing that out in the doc comment.

These are the things I wonder about since I’ve been working with syntax a lot recently 😉

Reply:
Yeah it's the body, not the syntax. I'll add a comment :)